packages/date-time-utilities/src/dateGrid/DateGrid.ts

```diff
@@ -0,0 +1,184 @@
+    weeksToShow,
+    workWeekDays,
+    daysToSelectInDayView,
+    restrictedDates,
+  } = options;
+  const restrictedDateOptions = { minDate, maxDate, restrictedDates };
+
+  const todaysDate = today || new Date();
+
+  const navigatedDate = options.navigatedDate ? options.navigatedDate : todaysDate;
+
+  let date;
+  if (weeksToShow && weeksToShow <= 4) {
+    // if showing less than a full month, just use date == navigatedDate
+    date = new Date(navigatedDate.toString());
+  } else {
+    date = new Date(navigatedDate.getFullYear(), navigatedDate.getMonth(), 1);
+  }
+  const weeks: IDay[][] = [];
+
+  // Cycle the date backwards to get to the first day of the week.
+  while (date.getDay() !== firstDayOfWeek) {
+    date.setDate(date.getDate() - 1);
+  }
+
+  // add the transition week as last week of previous range
+  date = addDays(date, -DAYS_IN_WEEK);
+
+  // a flag to indicate whether all days of the week are outside the month
+  let isAllDaysOfWeekOutOfMonth = false;
+
+  // in work week view if the days aren't contiguous we use week view instead
+  const selectedDateRangeType = getDateRangeTypeToUse(dateRangeType, workWeekDays);
+
+  let selectedDates = getDateRangeArray(
+    selectedDate,
+    selectedDateRangeType,
+    firstDayOfWeek,
+    workWeekDays,
+    daysToSelectInDayView,
+  );
+  selectedDates = getBoundedDateRange(selectedDates, minDate, maxDate);
+
+  let shouldGetWeeks = true;
+
+  for (let weekIndex = 0; shouldGetWeeks; weekIndex++) {
+    const week: IDay[] = [];
+
+    isAllDaysOfWeekOutOfMonth = true;
+
+    for (let dayIndex = 0; dayIndex < DAYS_IN_WEEK; dayIndex++) {
+      const originalDate = new Date(date.toString());
+      const dayInfo: IDay = {
+        key: date.toString(),
+        date: date.getDate().toString(),
+        originalDate: originalDate,
+        isInMonth: date.getMonth() === navigatedDate.getMonth(),
+        isToday: compareDates(todaysDate, date),
+        isSelected: isInDateRangeArray(date, selectedDates),
+        isInBounds: !isRestrictedDate(date, restrictedDateOptions),
+      };
+
+      week.push(dayInfo);
+
+      if (dayInfo.isInMonth) {
+        isAllDaysOfWeekOutOfMonth = false;
+      }
+
+      date.setDate(date.getDate() + 1);
+    }
+
+    // We append the condition of the loop depending upon the showSixWeeksByDefault prop.
+    shouldGetWeeks = weeksToShow ? weekIndex < weeksToShow + 1 : !isAllDaysOfWeekOutOfMonth || weekIndex === 0;
+
+    // we don't check shouldGetWeeks before pushing because we want to add one extra week for transition state
+    weeks.push(week);
+  }
+
+  return weeks;
+};
+
+export const isRestrictedDate = (date: Date, options: IRestrictedDatesOptions): boolean => {
+  const { restrictedDates, minDate, maxDate } = options;
+  if (!restrictedDates && !minDate && !maxDate) {
+    return false;
+  }
+  const inRestrictedDates =
+    restrictedDates &&
+    !!find(restrictedDates, (rd: Date) => {
+      return compareDates(rd, date);
+    });
```

Comment:
You can get rid of the uifabric/utilities reference here too (also in this case, switching to a more idiomatic way to achieve the desired result):
```suggestion
    restrictedDates.some((rd: Date) => compareDates(rd, date));
```